app/models/status_message.rb

```diff
@@ -120,6 +89,29 @@ def post_location
     }
   end
 
+  def people_allowed_to_be_mentioned
+    @aspects_ppl ||=
+      if public?
+        "all"
+      else
+        [
+          *AspectMembership.where(aspect_id: aspects).map(&:person),
+          *share_visibilities.map {|sv| sv.user.person }
```

Comment:
Oh, I missed the use in the `Notifications::Mentioned` ...

But I think it is better to create its own filter there, and only test if the person can see the post there (and for comments if the person can see the parent post)?